### PR TITLE
Katie/fix translation events table

### PR DESCRIPTION
**Pull request by KatieShipley**
Our translation status service is erroneously reporting some strings as not having translations, even though their translations have existed for a long time. This is because some translations have a period character in their string key. For example, `data -> blocks -> Dancelab_changeMoveEachLR -> options -> MOVE -> MOVES.Kick` is normalized to `data.blocks.Dancelab_changeMoveEachLR.options.MOVE.MOVES.Kick`. So `I18n.exists?` tries to find `MOVE -> MOVES -> Kick` (which does not exist) rather than `MOVE -> MOVES.KICK` (which we should be looking for).

Let's utilize the `string_key` and `scope` separately when checking if translations exist, like we do when actually translating the strings.

## Links

- jira ticket: [FND-1793](https://codedotorg.atlassian.net/browse/FND-1793)

### CODE REVIEW

File: bin/i18n/test/translation_status/test_translation_service.rb
Comment (hacodeorg) [inline bin/i18n/test/translation_status/test_translation_service.rb]: You could set empty string as the default value for `scope` to make the code a bit cleaner.

File: bin/i18n/translation_status/translation_service.rb
Comment (hacodeorg) [inline bin/i18n/translation_status/translation_service.rb]: Consider using `I18n.exists?` with fallback disabled https://github.com/ruby-i18n/i18n/pull/482

Comment (KatieShipley) [reply bin/i18n/translation_status/translation_service.rb]: Unfortunately we're on i18n v1.6.0, and that PR wasn't merged until v1.8.3. Can we get a ticket in the backlog to upgrade our i18n version and note that we should disable fallback here when that work is completed? Or do we want to block this work on that upgrade?

File: bin/i18n/test/translation_status/test_translation_service.rb
Comment (KatieShipley) [reply bin/i18n/test/translation_status/test_translation_service.rb]: Yes, good catch!

File: bin/i18n/translation_status/translation_service.rb
Comment (KatieShipley) [reply bin/i18n/translation_status/translation_service.rb]: Per slack discussion we've decided to upgrade and disable fallback after this merges

File: bin/i18n/translation_status/translation_service.rb
Comment (daynew) [inline bin/i18n/translation_status/translation_service.rb]: *nit pick* - Consider making this a proper if/else since it is kind of long.

File: lib/cdo/i18n_string_url_tracker.rb
Comment (daynew) [inline lib/cdo/i18n_string_url_tracker.rb]: Need to add the new parameters to the documentation

File: lib/cdo/i18n_backend.rb
Comment (daynew) [inline lib/cdo/i18n_backend.rb]: `normalized` key seems like it could be constructed inside `I18nStringUrlTracker.instance.log` rather than passed in as a parameter. Is there a reason we want the callers to construct it and pass it in?

File: lib/cdo/i18n_string_url_tracker.rb
Comment (daynew) [inline lib/cdo/i18n_string_url_tracker.rb]: Each of these are a separate column in the database, right? 
```suggestion
    # log every <url>:<normalized_key>:<source>:<string_key>:<scope>:<separator> combination to Firehose
```

File: lib/cdo/i18n_string_url_tracker.rb
Comment (daynew) [inline lib/cdo/i18n_string_url_tracker.rb]: same as above
```suggestion
          # record the <url>:<normalized_key>:<source>:<string_key>:<scope>:<separator> association.
```

File: lib/test/cdo/test_i18n_string_url_tracker.rb
Comment (daynew) [inline lib/test/cdo/test_i18n_string_url_tracker.rb:53]: Should we add a unit test for the behavior if any of the new parameters are null?

**Review by daynew — APPROVED** [review]
Mostly good. Just some comments I think you can resolve without needing another review.

File: lib/cdo/i18n_backend.rb
Comment (KatieShipley) [reply lib/cdo/i18n_backend.rb]: Nope no reason, that's just how it used to be so I hadn't thought to change it. Will be easy enough to move

**Pull request merged**
